flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java

```diff
@@ -69,4 +74,30 @@ public static Object convertConstant(Type type, Object value) {
     }
     return value;
   }
+
+  /**
+   * Similar to the private {@link RowDataSerializer#copyRowData(RowData, RowData)} method.
+   * This skips the check the arity of rowType and from,
+   * because the from RowData may contains additional column for position deletes.
+   * Using {@link RowDataSerializer#copy(RowData, RowData)} will fail the arity check.
+   */
+  public static RowData clone(RowData from, RowData reuse, RowType rowType, TypeSerializer[] fieldSerializers) {
```

Comment:
How the FLIP-27 use this method ?  How did they construct their `TypeSerializer` ?

Reply:
For my understanding,  this `clone` methods is really not friendly for developers to use.  If we really need to introduce a copy with checking the length, then how about making this to be private and expose an more easy method to public : 

```java
public static RowData clone(RowData from, RowType rowType)
```

Reply:
It is used here: https://github.com/stevenzwu/iceberg/blob/flip27IcebergSource/flink/src/main/java/org/apache/iceberg/flink/source/reader/RowDataIteratorBulkFormat.java#L116

The main reason for adding `TypeSerializer` to the `RowDataUtil#clone()` method is to avoid constructing it for each clone call. In the constructor of `RowDataIteratorBulkFormat`, we construct `TypeSerializer` once from `RowType`.

Reply:
OK ,  seems we're clone the rowData iterately,  I saw the  `FieldGetter` will be created for each row  [here](https://github.com/apache/iceberg/pull/2047/files#diff-81028bf7f8758383a5976b7c1461931a41485de740642252f5fd0c5c9cec51dfR94),  that should also not be the expected behavior.  We may need to introduce an new `RowDataCloner` which will initialize all its TypeSerializer &  `FieldGetter` once in instance constructor,  when iterating the `RowData` we will just clone row by row don't need to create any extra instances.

Reply:
I think it could be do the refactor when we review the [RowDataIteratorBulkFormat](https://github.com/stevenzwu/iceberg/blob/flip27IcebergSource/flink/src/main/java/org/apache/iceberg/flink/source/reader/RowDataIteratorBulkFormat.java#L116).